most_confident_indices/src/lib.rs

```diff
@@ -1,194 +1,98 @@
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::I16 => preprocess_buffer::<i16>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::U32 => preprocess_buffer::<u32>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::I32 => preprocess_buffer::<i32>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::F32 => preprocess_buffer::<f32>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::U64 => preprocess_buffer::<u64>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::I64 => preprocess_buffer::<i64>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::F64 => preprocess_buffer::<f64>(&buffer, &dimensions)
-                .and_then(|t| most_confident_indices(t, count))?,
-            ElementType::Utf8 => {
-                unreachable!("Already checked by get_element_type()")
+    fn run(&self, inputs: Vec<Tensor>) -> Result<Vec<Tensor>, RunError> {
+        let tensor = Tensor::get_named(&inputs, "confidences")?;
+
+        let count = parse::required_arg(&vec![Argument{ name: "count".to_string(), value: self.count.to_string() }], "count").unwrap(); 
+
+        let indices = match tensor.element_type {
+            ElementType::U8 => most_confident_indices(tensor.view_1d::<u8>()?, count)?,
+            ElementType::I8 => most_confident_indices(tensor.view_1d::<i8>()?, count)?,
+            ElementType::U16 => most_confident_indices(tensor.view_1d::<u16>()?, count)?,
+            ElementType::I16 =>most_confident_indices(tensor.view_1d::<i16>()?, count)?,
+            ElementType::U32 => most_confident_indices(tensor.view_1d::<u32>()?, count)?,
+            ElementType::I32 => most_confident_indices(tensor.view_1d::<i32>()?, count)?,
+            ElementType::F32 => most_confident_indices(tensor.view_1d::<f32>()?, count)?,
+            ElementType::U64 => most_confident_indices(tensor.view_1d::<u64>()?, count)?,
+            ElementType::I64 => most_confident_indices(tensor.view_1d::<i64>()?, count)?,
+            ElementType::F64 => most_confident_indices(tensor.view_1d::<f64>()?, count)?,
+            _ => {
+                return Err(InvalidInput::incompatible_element_type(
+                    "confidences",
+                )
+                .into());
             },
         };
 
-        ctx.set_output_tensor(
-            "indices",
-            TensorParam {
-                dimensions: &dimensions,
-                element_type: ElementType::U32,
-                buffer: indices.as_bytes(),
-            },
-        );
-
-        Ok(())
+        Ok(vec![Tensor::new_1d("indices", &indices.to_vec())])
```

Comment:
The `.to_vec()` here is redundant. The `indices` variable has type `Vec<u32>`, and doing `&indices` will pass a `&[u32]` to the `Tensor::new_1d()` function[^1].

```suggestion
        Ok(vec![Tensor::new_1d("indices", &indices)])
```

[^1]: Well technically `&indices` gives us a `&Vec<u32>`, but `Vec<u32>` implements `Deref` where the `Target` type is `[u32]`, so the compiler can automatically coerce `&Vec<u32>` to `&[u32]` because of "deref coersions".